pkg/cosign/rego/rego_test.go

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rego
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+const simpleJSONBody = `{
+	"_type": "https://in-toto.io/Statement/v0.1",
+	"predicateType": "https://slsa.dev/provenance/v0.2"
+}`
+
+func TestValidationJSON(t *testing.T) {
+	cases := []struct {
+		name     string
+		jsonBody string
+		policy   string
+		pass     bool
+		errors   []string
+	}{
+		{
+			name:     "passing policy",
+			jsonBody: simpleJSONBody,
+			policy: `
+				package signature
+
+				allow {
+					input.predicateType == "https://slsa.dev/provenance/v0.2"
+				}
+			`,
+			pass: true,
+		},
+		{
+			name:     "undefined result due to no matching rules",
+			jsonBody: simpleJSONBody,
+			policy: `
+				package signature
+
+				allow {
+					input.predicateType == "https://slsa.dev/provenance/v99.9"
+				}
+			`,
+			pass:   false,
+			errors: []string{"result is undefined for query 'data.signature.allow'"},
+		},
+		{
+			name:     "policy query evaluates to false",
+			jsonBody: simpleJSONBody,
+			policy: `
+				package signature
+
+				default allow = false
+			`,
+			pass:   false,
+			errors: []string{"expression value, false, is not true"},
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			policyFileName := filepath.Join(t.TempDir(), "policy.rego")
+			if err := os.WriteFile(policyFileName, []byte(tt.policy), 0644); err != nil {
```

Comment:
On Windows, `os.WriteFile` doesn't return an error (good), but when `rego.ValidateJSON` tries to read the file, Windows complains it's not there:
```
CreateFile \Users\RUNNER~1\AppData\Local\Temp\TestValidationJSONpassing_policy1204380751\001\policy.rego: The system cannot find the path specified.
```

*Even though the Windows call is `CreateFile`, I believe it's actually used for both reading and writing files.

There are other parts of the code that follow this same pattern and the tests seem to work just fine. I got my hands on a windows machine and will try to reproduce this locally.

Reply:
I cannot reproduce this locally with golang 1.17.8 (same used in tests) on Windows 11. I could disable these new tests for Windows but that is not an ideal solution.

@dlorenc, any suggestions?

Reply:
This is an issue on rego itself: https://github.com/open-policy-agent/opa/issues/4521

Adjusted the tests to take this into account.